### ISSUE TITLE
EVP_get_cipherbyname(): inconsistent cipher name case

### ISSUE THREAD

**Issue by davidebeatrici** [openssl/openssl#6921]
```c
EVP_get_cipherbyname("aes-128-cbc"); // Returns an EVP_CIPHER structure
EVP_get_cipherbyname("AES-128-CBC"); // Returns an EVP_CIPHER structure
EVP_get_cipherbyname("aes-128-gcm"); // Returns an EVP_CIPHER structure
EVP_get_cipherbyname("AES-128-GCM"); // Returns NULL
```

On IRC (`##openssl@freenode`), @mancha1 mentioned that the alternate name for `aes-128-gcm` is `id-aes128-GCM`, according to [RFC5084](https://tools.ietf.org/html/rfc5084).

I think that you could either:
- Add `AES-128-GCM` as an alternate name (and do the same with the other affected ciphers);
- Mention on the Docs that `EVP_get_cipherbyname()` uses the RFC5084 naming.

**Comment by levitte** [comment]
I believe that we following RFC naming for *all* our cipher names, but the diverse RFCs aren't always consistent (I suppose that there are historical reasons).

**Comment by t-j-h** [comment]
I don't think it would hurt to be doing a case insensitive comparison for these.

**Comment by levitte** [comment]
Yeah, you're right about that. That would be to be done in the OBJ_NAME functions....

**Comment by paulidale** [comment]
I agree that a case insensitive comparison would be useful.

**Comment by davidebeatrici** [comment]
Excellent, thank you very much!
